model/src/main/java/org/projectnessie/model/IcebergTable.java

```diff
@@ -18,21 +18,80 @@
 import com.fasterxml.jackson.annotation.JsonTypeName;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import javax.annotation.Nullable;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
 import org.immutables.value.Value;
 
+/**
+ * Represents the state of an Iceberg table in Nessie. An Iceberg table is globally identified via
+ * its {@link Contents#getId() unique ID}.
+ *
+ * <p>The Iceberg-table-state consists of the location to the table-metadata and the snapshot-ID.
+ *
+ * <p>The table-metadata-location is managed globally within Nessie, which means that all versions
+ * of the same table share the table-metadata across all named-references (branches and tags). There
+ * is only one version, the current version, of Iceberg's table-metadata.
```

Comment:
nit: Would it make sense to emphasize that the global metadata is conceptually cumulative (contains information that applies equally to all tables, current and historic)?

Reply:
Not sure though. It may change in the future.

Reply:
However, even in general from Nessies' perspective it probably only makes sense to track something as "global" state only if it is cumulative in nature, right? Otherwise, it'd be part of the change/commit log, would it not?